src/bun.js/bindings/webcore/FetchHeaders.cpp

```diff
@@ -261,30 +268,56 @@ void FetchHeaders::filterAndFill(const HTTPHeaderMap& headers, Guard guard)
     }
 }
 
-static NeverDestroyed<const String> setCookieLowercaseString(MAKE_STATIC_STRING_IMPL("set-cookie"));
+static bool compareIteratorKeys(const String& a, const String& b)
+{
+    // null in the iterator's m_keys represents Set-Cookie.
+    return WTF::codePointCompareLessThan(
+        a.isNull() ? "set-cookie"_s : a,
+        b.isNull() ? "set-cookie"_s : b);
+}
 
 std::optional<KeyValuePair<String, String>> FetchHeaders::Iterator::next()
 {
     if (m_keys.isEmpty() || m_updateCounter != m_headers->m_updateCounter) {
+        bool hasSetCookie = !m_headers->getSetCookieHeaders().isEmpty();
         m_keys.resize(0);
-        m_keys.reserveCapacity(m_headers->m_headers.size());
+        m_keys.reserveCapacity(m_headers->m_headers.size() + (hasSetCookie ? 1 : 0));
         for (auto& header : m_headers->m_headers)
             m_keys.uncheckedAppend(header.asciiLowerCaseName());
-        std::sort(m_keys.begin(), m_keys.end(), WTF::codePointCompareLessThan);
+        if (hasSetCookie)
```

Comment:
Do we need to pay this performance overhead here? Wouldn't it be fine if we put the Set-Cookie header either first or last?

Reply:
we do, this is part of the fetch spec https://fetch.spec.whatwg.org/#convert-header-names-to-a-sorted-lowercase-set

Reply:
It was also expected in the spec for keys to be unique which isn’t true for set-cookies 

I think we should just always put it at the end and call it a day. Each call to next() shouldn’t slow it down

Reply:
updated to append after the sort